Added test & fixed a bug in cloneRepo

beginner003 opened the pull request:
- Tests implemented to test for a local server and simulated the payload delivery by a customised JSON payload object, passed
- Tests implemented to test the payload parsing (the ref, commit hash, repo url, and branch name), passed
- Tests implemented to test for the function cloneRepo() in the tmp directory
- Bug fixed in the cloneRepo()
- dependency for server testing is added to pom.xml

beginner003 committed:
Added test & fixed a bug in cloneRepo

- Tests implemented to test for a local server and simulated the payload delivery by a customised JSON payload object, passed
- Tests implemented to test the payload parsing (the ref, commit hash, repo url, and branch name), passed
- Tests implemented to test for the function cloneRepo() in the tmp directory
- Bug fixed in the cloneRepo()
- dependency for server testing is added to pom.xml

edgarwolff commented:
Switched to draft to add automated resource setup

beginner003 committed:
Fixed test format to be automatic

- Amended the test to have automatic repo creation and hash and url retrieval.
- Created the sh file containing the commands to create the local repo
- passed the test

beginner003 commented:
> Switched to draft to add automated resource setup

Test automated for server and repo setup in the commit: [Fixed test format to be automatic](https://github.com/DD2480-Group-27/assignment2-CI/pull/21/commits/4d3d02e14db72ef9ae05e4fd4e8b2fe94eeeaea8)

edgarwolff commented on pom.xml:
```diff
@@ -47,6 +47,17 @@
+        <dependency>
+            <groupId>ch.qos.logback</groupId>
+            <artifactId>logback-classic</artifactId>
+            <version>1.3.0</version> <!-- ensure this is compatible -->
```
Remove such comments

edgarwolff commented on src/main/java/CIServer.java:
```diff
@@ -37,7 +39,7 @@ public void handle(String target,
         }
 
-        // System.out.println("Payload received:\n" + payload.toString());
+         System.out.println("Payload received:\n" + payload.toString());
```
Do we need to print the payload?

edgarwolff commented on src/main/java/CIServer.java:
```diff
@@ -116,10 +123,23 @@ private String cloneRepo(String repoURL, String commitHash, String branchName) {
+        directory.delete(); }
     // used to start the CI server in command line
     public static void main(String[] args) throws Exception {
         Server server = new Server(8027);
```
If executing the tests on the CI server, can this interfeer with the port already open on the CI server?

edgarwolff replied:
I can tell it does after testing locally

beginner003 replied:
Solved!

edgarwolff replied:
I can tell not by testing it

edgarwolff reviewed (changes requested):
I cannot accept this PR because way to many debug informations are getting printed to the console, and they are hidding multiple errors that are also getting printed in the console but are almost hidden in all this textual mess
It would be relevant to hide if not needed the DEBUG informations and check why the errors are happening if we are testing a correct value in the system.
I also saw that this PR modifies the code from CI server, is there a functionnal change in the code? Is the code in the main branch correct?
Your code needs to be correct in order to merge the codeverification branch as well as start the email notification (which relies on the parts before) by Kristin.
Remeber deadline is tomorrow, it is starting to be urgent.

Btw some complete documentation for the method cloneRepo and handle would be nice

beginner003 committed:
Fixed code issues, added documentation in CIServerTest.java

- Added a port checking method to determine if a new server should be started or not.
- Removed debugging printIn statements & logging dependencies in pom.xml
- Added documentation for helper function
- Fixed the typo in server_setup.sh (removed the line cd ...)

beginner003 committed:
Added documentation and removed unnecessary logging statements

- added description for all the methods
- removed some of the logging statements.

beginner003 commented:
> I cannot accept this PR because way to many debug informations are getting printed to the console, and they are hidding multiple errors that are also getting printed in the console but are almost hidden in all this textual mess It would be relevant to hide if not needed the DEBUG informations and check why the errors are happening if we are testing a correct value in the system. I also saw that this PR modifies the code from CI server, is there a functionnal change in the code? Is the code in the main branch correct? Your code needs to be correct in order to merge the codeverification branch as well as start the email notification (which relies on the parts before) by Kristin. Remeber deadline is tomorrow, it is starting to be urgent.
> 
> Btw some complete documentation for the method cloneRepo and handle would be nice

Check the two new commits. [Fixed code issues, added documentation in CIServerTest.java](https://github.com/DD2480-Group-27/assignment2-CI/pull/21/commits/904ccf884b99556e90462b64e6e9f533a8158a09) [Added documentation and removed unnecessary logging statements](https://github.com/DD2480-Group-27/assignment2-CI/pull/21/commits/94a92136d9d73f6bdac1e8200cc40e5ea9e81bf6).
These two commits fixed all the requests, including documentation, removing logging statements, server starting conflicts.
The funcitonal change in the CIServer.java is mentioned in this commit [Added test & fixed a bug in cloneRepo](https://github.com/DD2480-Group-27/assignment2-CI/pull/21/commits/8b230e4e06915b151007597ad6b296478dd92210). It is a bug in the cloneRepo() method (the parsing of the branch name ref/heads (missing s in the previous version).

edgarwolff commented on src/main/java/CIServer.java:
```diff
@@ -14,20 +14,45 @@
+                       HttpServletResponse response)
+        throws IOException, ServletException
     {
+//        System.out.println("In handle process");
```
Debugging code should not be pushed towards main branch (I assume this is here for debugging purposes to check if the method was indeed called when doing integratino tests locally)

edgarwolff reviewed (changes requested):
Last review's comment are still not covered

edgarwolff committed:
refact: Change code structure to enable efficient unit testing

Refactored the CIServer.java file in adding an imbricated calss to parse the needed information from webhook JSON data
Refactored the CIServerTest.java file to match the new implementation in CIServer.java

beginner003 committed:
Refactored the code in CIServer.java.

- separate the class WebhookJSONAnalyser from CIServer for maintainable structure.

beginner003 committed:
Refactored the code in CIServerTest.java and created test samples

- separate the test for WebhookJSONAnalyser from the CIServer.
- Create test samples:
Steps Taken:
  1.Webhook Creation: A webhook was set up to listen for events from the Lab 1 repository.
  2.Empty Commits: Two empty commits were made:
  One on the main branch.
  Another on the branch named kristin/doc/readme-add-essence.
  3.Payload Data: The payload data received from the webhook upon these commits has been copied and pasted below for further analysis and testing.
- Created unit tests for the samples to test if WebhookJSONAnalyser.java is retrieving correct data and if the repo is located at the right place.

beginner003 committed:
Fixed bug in WebhookJSONAnalyser.java

- the git checking out is fixed to adapt to more general branch names.

blimpan reviewed (approved):
Works on remote machine. I think we have tested what we can.
